pkg/virt-controller/watch/network_test.go

```diff
@@ -261,6 +262,81 @@ var _ = Describe("Network interface hot{un}plug", func() {
 			[]v1.VirtualMachineInterfaceRequest{},
 		),
 	)
+
+	Context("handle dynamic interface requests", func() {
+		var clusterConfig *virtconfig.ClusterConfig
+
+		BeforeEach(func() {
+			clusterConfig, _, _ = testutils.NewFakeClusterConfigUsingKVConfig(&v1.KubeVirtConfiguration{})
+		})
+
+		DescribeTable("for a VM with no interfaces",
+			func(requests []v1.VirtualMachineInterfaceRequest, expectedInterfaces []v1.Interface, vmi *v1.VirtualMachineInstance) {
+				vm := kubecli.NewMinimalVM("test")
+				vm.Spec.Template = &v1.VirtualMachineInstanceTemplateSpec{}
+				vm.Status.InterfaceRequests = requests
+
+				Expect(handleDynamicInterfaceRequests(vm, vmi, clusterConfig)).To(Succeed())
+				Expect(vm.Spec.Template.Spec.Domain.Devices.Interfaces).To(Equal(expectedInterfaces))
```

Comment:
For completion, I think we also need to check the networks.

Reply:
done.